### ISSUE TITLE
`bundle exec jekyll build` results in exception, nothing generated

### ISSUE THREAD

**Issue by DavidBiesack** [launchany/mvp-template#3]
I cloned this repo, ran `bundle install`, then tried to build the site so I could preview it on localhost, but it fails with a non-ASCII character in _layouts
```
$  bundle exec jekyll build
Cannot read termcap database;
using dumb terminal settings.
Configuration file: /private/tmp/mvd-template/_config.yml
            Source: /private/tmp/mvd-template
       Destination: /private/tmp/mvd-template/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
  Liquid Exception: invalid byte sequence in US-ASCII in _layouts/redirect.html
jekyll 3.4.3 | Error:  invalid byte sequence in US-ASCII
```
Unfortunately, `_layouts/redirect.html` is not in this repo 
```
$ ls _layouts
about.html      default.html    docs.html       page.html       post.html
```
and I don't know where it comes from.

Running with `--verbose --trace`:
```
$    bundle exec jekyll build --verbose --trace
Cannot read termcap database;
using dumb terminal settings.
  Logging at level: debug
Configuration file: /private/tmp/mvd-template/_config.yml
      GitHub Pages: github-pages v141
      GitHub Pages: jekyll v3.4.3
         Requiring: jekyll-feed
         Requiring: jekyll-redirect-from
         Requiring: jekyll-seo-tag
         Requiring: jekyll-sitemap
         Requiring: jekyll-coffeescript
         Requiring: jekyll-gist
         Requiring: jekyll-github-metadata
Dotenv not found. Skipping 
         Requiring: jekyll-paginate
         Requiring: jekyll-relative-links
         Requiring: jekyll-optional-front-matter
         Requiring: jekyll-readme-index
         Requiring: jekyll-default-layout
         Requiring: jekyll-titles-from-headings
         Requiring: kramdown
   GitHub Metadata: Initializing...
            Source: /private/tmp/mvd-template
       Destination: /private/tmp/mvd-template/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       EntryFilter: excluded /README.md
       EntryFilter: excluded /Gemfile
       EntryFilter: excluded /Gemfile.lock
           Reading: _posts/2017-05-03-jekyll-doc-theme.md
           Reading: _posts/2017-02-01-welcome-to-jekyll.md
           Reading: _docs/workflows.md
           Reading: _docs/code_samples.md
           Reading: _docs/mvd_checklist.md
           Reading: _docs/index.md
           Reading: _docs/mdreference.md
           Reading: _docs/openapi.md
           Reading: _docs/publishing_options.md
           Reading: _about/index.md
        Generating: JekyllRedirectFrom::Generator finished in 0.015146 seconds.
        Generating: JekyllOptionalFrontMatter::Generator finished in 4.1e-05 seconds.
        Generating: JekyllReadmeIndex::Generator finished in 0.000107 seconds.
        Generating: JekyllRelativeLinks::Generator finished in 1.8e-05 seconds.
        Generating: JekyllDefaultLayout::Generator finished in 4.6e-05 seconds.
        Generating: JekyllFeed::Generator finished in 0.000853 seconds.
        Generating: JekyllTitlesFromHeadings::Generator finished in 1.5e-05 seconds.
        Generating: Jekyll::JekyllSitemap finished in 0.000318 seconds.
        Generating: Jekyll::Paginate::Pagination finished in 7.0e-06 seconds.
         Rendering: _posts/2017-02-01-welcome-to-jekyll.md
  Pre-Render Hooks: _posts/2017-02-01-welcome-to-jekyll.md
  Rendering Liquid: _posts/2017-02-01-welcome-to-jekyll.md
         Requiring: rouge
  Rendering Markup: _posts/2017-02-01-welcome-to-jekyll.md
         Requiring: kramdown
  Rendering Layout: _posts/2017-02-01-welcome-to-jekyll.md
         Rendering: _posts/2017-02-01-welcome-to-jekyll.md
  Pre-Render Hooks: _posts/2017-02-01-welcome-to-jekyll.md
  Rendering Liquid: _posts/2017-02-01-welcome-to-jekyll.md
  Rendering Markup: _posts/2017-02-01-welcome-to-jekyll.md
         Rendering: _posts/2017-05-03-jekyll-doc-theme.md
  Pre-Render Hooks: _posts/2017-05-03-jekyll-doc-theme.md
  Rendering Liquid: _posts/2017-05-03-jekyll-doc-theme.md
  Rendering Markup: _posts/2017-05-03-jekyll-doc-theme.md
  Rendering Layout: _posts/2017-05-03-jekyll-doc-theme.md
         Rendering: _posts/2017-05-03-jekyll-doc-theme.md
  Pre-Render Hooks: _posts/2017-05-03-jekyll-doc-theme.md
  Rendering Liquid: _posts/2017-05-03-jekyll-doc-theme.md
  Rendering Markup: _posts/2017-05-03-jekyll-doc-theme.md
         Rendering: _docs/code_samples.md
  Pre-Render Hooks: _docs/code_samples.md
  Rendering Liquid: _docs/code_samples.md
  Rendering Markup: _docs/code_samples.md
  Rendering Layout: _docs/code_samples.md
         Rendering: _docs/code_samples.md
  Pre-Render Hooks: _docs/code_samples.md
  Rendering Liquid: _docs/code_samples.md
  Rendering Markup: _docs/code_samples.md
         Rendering: _docs/index.md
  Pre-Render Hooks: _docs/index.md
  Rendering Liquid: _docs/index.md
  Rendering Markup: _docs/index.md
  Rendering Layout: _docs/index.md
         Rendering: _docs/index.md
  Pre-Render Hooks: _docs/index.md
  Rendering Liquid: _docs/index.md
  Rendering Markup: _docs/index.md
         Rendering: _docs/mdreference.md
  Pre-Render Hooks: _docs/mdreference.md
  Rendering Liquid: _docs/mdreference.md
  Rendering Markup: _docs/mdreference.md
  Rendering Layout: _docs/mdreference.md
         Rendering: _docs/mdreference.md
  Pre-Render Hooks: _docs/mdreference.md
  Rendering Liquid: _docs/mdreference.md
  Rendering Markup: _docs/mdreference.md
         Rendering: _docs/mvd_checklist.md
  Pre-Render Hooks: _docs/mvd_checklist.md
  Rendering Liquid: _docs/mvd_checklist.md
  Rendering Markup: _docs/mvd_checklist.md
  Rendering Layout: _docs/mvd_checklist.md
         Rendering: _docs/mvd_checklist.md
  Pre-Render Hooks: _docs/mvd_checklist.md
  Rendering Liquid: _docs/mvd_checklist.md
  Rendering Markup: _docs/mvd_checklist.md
         Rendering: _docs/openapi.md
  Pre-Render Hooks: _docs/openapi.md
  Rendering Liquid: _docs/openapi.md
  Rendering Markup: _docs/openapi.md
  Rendering Layout: _docs/openapi.md
         Rendering: _docs/openapi.md
  Pre-Render Hooks: _docs/openapi.md
  Rendering Liquid: _docs/openapi.md
  Rendering Markup: _docs/openapi.md
         Rendering: _docs/publishing_options.md
  Pre-Render Hooks: _docs/publishing_options.md
  Rendering Liquid: _docs/publishing_options.md
  Rendering Markup: _docs/publishing_options.md
  Rendering Layout: _docs/publishing_options.md
         Rendering: _docs/publishing_options.md
  Pre-Render Hooks: _docs/publishing_options.md
  Rendering Liquid: _docs/publishing_options.md
  Rendering Markup: _docs/publishing_options.md
         Rendering: _docs/workflows.md
  Pre-Render Hooks: _docs/workflows.md
  Rendering Liquid: _docs/workflows.md
  Rendering Markup: _docs/workflows.md
  Rendering Layout: _docs/workflows.md
         Rendering: _docs/workflows.md
  Pre-Render Hooks: _docs/workflows.md
  Rendering Liquid: _docs/workflows.md
  Rendering Markup: _docs/workflows.md
         Rendering: _about/index.md
  Pre-Render Hooks: _about/index.md
  Rendering Liquid: _about/index.md
  Rendering Markup: _about/index.md
  Rendering Layout: _about/index.md
         Rendering: _about/index.md
  Pre-Render Hooks: _about/index.md
  Rendering Liquid: _about/index.md
  Rendering Markup: _about/index.md
         Rendering: 404.html
  Pre-Render Hooks: 404.html
  Rendering Liquid: 404.html
  Rendering Markup: 404.html
  Rendering Layout: 404.html
         Rendering: about.html
  Pre-Render Hooks: about.html
  Rendering Liquid: about.html
  Rendering Markup: about.html
  Rendering Layout: about.html
         Rendering: allposts.html
  Pre-Render Hooks: allposts.html
  Rendering Liquid: allposts.html
  Rendering Markup: allposts.html
  Rendering Layout: allposts.html
         Rendering: index.html
  Pre-Render Hooks: index.html
  Rendering Liquid: index.html
  Rendering Markup: index.html
  Rendering Layout: index.html
         Rendering: css/main.scss
  Pre-Render Hooks: css/main.scss
  Rendering Liquid: css/main.scss
  Rendering Markup: css/main.scss
         Rendering: search.json
  Pre-Render Hooks: search.json
  Rendering Liquid: search.json
  Rendering Markup: search.json
  Rendering Layout: search.json
         Rendering: redirect.html
  Pre-Render Hooks: redirect.html
  Rendering Liquid: redirect.html
  Rendering Markup: redirect.html
  Rendering Layout: redirect.html
  Liquid Exception: invalid byte sequence in US-ASCII in _layouts/redirect.html
bundler: failed to load command: jekyll (/usr/local/bin/jekyll)
ArgumentError: invalid byte sequence in US-ASCII
  /Library/Ruby/Gems/2.3.0/gems/liquid-3.0.6/lib/liquid/template.rb:232:in `split'
  /Library/Ruby/Gems/2.3.0/gems/liquid-3.0.6/lib/liquid/template.rb:232:in `tokenize'
  /Library/Ruby/Gems/2.3.0/gems/liquid-3.0.6/lib/liquid/template.rb:122:in `parse'
  /Library/Ruby/Gems/2.3.0/gems/liquid-3.0.6/lib/liquid/template.rb:108:in `parse'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/liquid_renderer/file.rb:11:in `block in parse'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/liquid_renderer/file.rb:47:in `measure_time'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/liquid_renderer/file.rb:10:in `parse'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/renderer.rb:129:in `render_liquid'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/renderer.rb:177:in `place_in_layouts'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/renderer.rb:91:in `run'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/site.rb:463:in `block in render_pages'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/site.rb:461:in `each'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/site.rb:461:in `render_pages'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/site.rb:191:in `render'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/site.rb:69:in `process'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/command.rb:26:in `process_site'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/commands/build.rb:63:in `build'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/commands/build.rb:34:in `process'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/lib/jekyll/commands/build.rb:16:in `block (2 levels) in init_with_program'
  /Library/Ruby/Gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
  /Library/Ruby/Gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
  /Library/Ruby/Gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
  /Library/Ruby/Gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
  /Library/Ruby/Gems/2.3.0/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
  /Library/Ruby/Gems/2.3.0/gems/jekyll-3.4.3/exe/jekyll:13:in `<top (required)>'
  /usr/local/bin/jekyll:22:in `load'
  /usr/local/bin/jekyll:22:in `<top (required)>'
```

In trying to diagnose, I commented out
```
   gem "jekyll-redirect-from"
```
from `Gemfile`, then tried again wit
```
bundle install
bundle exec jekyll build
```
which failed because the redirect package is needed. I uncommented it back and rebuilt
and this time it build cleanly. Looking at the install, the first time it picked up 1.12.1
but the second time, 
```
Using jekyll-redirect-from 0.14.0
```
So some dependency must be older and pull in 1.12.1 but when allowed to resolve on its own, picks up 1.14.0

I'd prefer a fix (in Gemfile?) so that I can push this build into our local CICD pipeline to automate the build/publication of the site and have it work reliably.

**Comment by DavidBiesack** [comment]
Commenting out 
```
gem 'github-pages', group: :jekyll_plugins
```
allows Gem to resolve to 1.14.0 but putting it back in results in a conflict:
```
$ bundle exec jekyll build --verbose
Bundler could not find compatible versions for gem "jekyll-redirect-from":
  In snapshot (Gemfile.lock):
    jekyll-redirect-from (= 0.14.0)

  In Gemfile:
    jekyll-redirect-from

    github-pages was resolved to 141, which depends on
      jekyll-redirect-from (= 0.12.1)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

**Comment by jhigginbotham** [comment]
I have pushed an updated Gemfile.lock with the latest dependencies. I was able to go through a successful build using Ruby 2.4.1 on MacOS. Let me know if this resolves the issue, or if there are other lingering issues preventing you from building.

**Comment by DavidBiesack** [comment]
I cloned this into `/tmp/mvd-template` but the build still failed:
```
 cd /tmp
git clone https://github.com/launchany/mvd-template.git
cd mvd-template
bundle install
bundle exec jekyll build
Configuration file: /private/tmp/mvd-template/_config.yml
            Source: /private/tmp/mvd-template
       Destination: /private/tmp/mvd-template/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
  Liquid Exception: invalid byte sequence in US-ASCII in _layouts/redirect.html
jekyll 3.4.3 | Error:  invalid byte sequence in US-ASCII
```

**Comment by DavidBiesack** [comment]
`git log` shows my local repo has your latest
`* fb6c01ce6c3a41bbdee7bb1bccf592bd370f14eb (HEAD -> gh-pages, origin/gh-pages, origin/HEAD) Updating rubygems to the latest versions of dependencies`

**Comment by jhigginbotham** [comment]
I'll need a bit more information to debug then. What OS? What version of Ruby?

**Comment by DavidBiesack** [comment]
```
$ uname -v
Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64
$ ruby --version
ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin17]
$ gem --version
2.5.2.3
```
OS X 10.13.6 (17G65)

**Comment by jhigginbotham** [comment]
I use rvm, which must compile ruby with some extensions to happily deal with unicode. I copied the redirect.html file into _layouts for this project, overriding the one with the unicode characters that cause the issue. I then rebuilt the project and verified that my fixed three-dot (...) version is being used instead of the ellipsis version in the gem. See if this resolves it for you, as I can't seem to reproduce on my side but I have confirmed this is an issue from here: https://github.com/jekyll/jekyll-redirect-from/issues/139
